app/routes/UserController.js

```diff
@@ -45,29 +47,53 @@ router.get("", (req, res) => {
 */
 
 // LOG-IN --> verb GET of HTTP, so I obtain an existing resource (CRUD operation: Read)
-router.get("", async (req, res) => {
+router.get("", async (req, res) => { 
+      
   try {
-    const username_in = req.body.username; // TODO: mettere controllo in input: se utente lascia campo vuoto, ....
+    const username_in = req.body.username;
     const password_in = req.body.password;
+
+    try {
+      validateLogin(username_in, password_in);
+    }catch (error) {
+      console.error(error.message); // Handling the error
+      return res.status(400).send(error.message);
+    }
+
     const user = await User.findOne({ username: username_in }); // N.B.: await allows us to write asynchronous code so that it seems synchronous.
     // the execution of the async function is paused until the Promise (here findOne()) became fulfilled (resolved) or rejected.
+
     // Worst case: User NOT found
-    if (!user || user.password != password_in) {
+    if (!user) {
       // HTTP error 401: indicates that the request has not been applied because it lacks valid authentication credentials for the target resource.
-      return res.status(401).send("Authentication Failed: User not Found"); // TODO: pwd sbagliata
+      return res.status(401).send("Authentication Failed: User not Found");
+    }
+
+    // Worst case: PWD do not match
+    if (user.password != password_in) {
+      return res.status(401).send("Authentication Failed: Password Uncorrect");
     }
-    // Better case: User found
-    // I generate a JWT token to securely transfer encrypted data between client and server.
-    // Client could also reuse this token to authenticate subsequent requests to the server.
+
+    // Better case: User found and password is correct
+    // I generate a JWT token to securely transfer encrypted data between client and server. Client could also reuse this token to authenticate subsequent requests to the server.
     const token = jwt.sign({ userId: user._id }, process.env.JWT_SECRET, {
       expiresIn: EXPIRAL_AUTH_TIME,
     }); // payload, secret, options
+
     // I send an HTTP response to the client with status 200 (request has succeeded) and the token in JSON format
     res.status(200).json({ token }); // Convenient and powerful way to handle HTTP responses in Express
-  } catch (error) {
+  }catch (error){
     // the server encountered an unexpected condition that prevented it from fulfilling the request. So I send the error message.
     res.status(500).send(error.message);
   }
 });
 
+
+router.get("/:id/Users", async (req, res) => {  // I use the parameter :id of the Alert to get the user with that specific id
```

Comment:
users in piccolo (mi sa)

Reply:
questa cosa va chiesta al prof, quindi perfavore mettere un commento che ce lo ricordi

Reply:
yes